### PR TITLE
Adapt to COCO format dataset

### PR DESCRIPTION
**Pull request by yyqgood**
It is not only adapted to the COCO dataset, but also to the dataset of the COCO format type.

### CODE REVIEW

**Comment by FateScript** [comment]
Please lint your code to pass CI.

**Comment by yyqgood** [comment]
Code checked

File: yolox/evaluators/coco_evaluator.py [hunk 1]
```diff
@@ -27,9 +27,33 @@
 )
 
 
+def per_class_AR_table(coco_eval, class_names=COCO_CLASSES, headers=["class", "AR"], colums=6):
+    per_class_AR = {}
+    recalls = coco_eval.eval["recall"]
+    # dimension of recalls: [TxKxAxM]
+    # precision has dims (iou, cls, area range, max dets)
```
Comment (FateScript) [inline yolox/evaluators/coco_evaluator.py:34]: precision -> recall?

File: yolox/evaluators/coco_evaluator.py [hunk 2]
```diff
@@ -27,9 +27,33 @@
 )
 
 
+def per_class_AR_table(coco_eval, class_names=COCO_CLASSES, headers=["class", "AR"], colums=6):
```
Comment (FateScript) [inline yolox/evaluators/coco_evaluator.py:30]: `per_class_AR_table is similar` to `per_calss_AP_table`, why not merge them into one function?

File: yolox/evaluators/coco_evaluator.py (hunk 1 above)
Comment (yyqgood) [reply yolox/evaluators/coco_evaluator.py:34]: precision points to AP; recall points to AR。
Precision is obtained using coco_eval.eval["precision"]; 
recall is obtained using coco_eval.eval["recall"]. 
The parameters for obtaining the results are defined in the class Params under pycocotools/cocoeval.py

File: yolox/evaluators/coco_evaluator.py (hunk 2 above)
Comment (yyqgood) [reply yolox/evaluators/coco_evaluator.py:30]: First. At that time, in order to quickly test AP and AR under different conditions, such as different IoU and different regional range indicators.
Second. I don't want to study the third-party library of tabulation too much.
Third. In actual projects, AP and AR are equally important, and the use of two functions reflects the same importance.
Of course, I may change some of the above thoughts later, but now it is very flexible to use. (Later, may be changed to AP, AR, create_table three functions)

File: yolox/evaluators/coco_evaluator.py (hunk 1 above)
Comment (FateScript) [reply yolox/evaluators/coco_evaluator.py:34]:
```suggestion
    # recall has dims (iou, cls, area range, max dets)
```
is what I mean.

Comment (yyqgood) [reply yolox/evaluators/coco_evaluator.py]: Modified, thank you

**Comment by FateScript** [comment]
LGTM.

File: yolox/evaluators/coco_evaluator.py [hunk 3]
```diff
@@ -254,7 +279,9 @@ def evaluate_prediction(self, data_dict, statistics):
                 cocoEval.summarize()
             info += redirect_string.getvalue()
             if self.per_class_mAP:
-                info += "per class mAP:\n" + per_class_mAP_table(cocoEval)
+                info += "per class mAP:\n" + per_class_mAP_table(cocoEval) + "\n"
+            if self.per_class_AR:
```
Comment (759401524) [inline yolox/evaluators/coco_evaluator.py:283]: AttributeError: 'COCOEvaluator' object has no attribute 'per_class_AR'

Comment (yyqgood) [reply yolox/evaluators/coco_evaluator.py:283]: #1065

Comment (FateScript) [reply yolox/evaluators/coco_evaluator.py:283]: @759401524 fixed now, please pull the latest branch.

**Pull request merged**